spaceWeatherApi/Utils/DataUtils.cs

```diff
@@ -0,0 +1,225 @@
+﻿
+using SpaceWeatherApi.DataModels;
+namespace SpaceWeatherApi.Utils
+{
+
+    public class DataUtils
+    {
+        
+
+        /// <summary>
+        /// Create a detailed report of correlated solar events
+        /// </summary>
+        /// <param name="flare"></param>
+        /// <param name="cme"></param>
+        /// <returns></returns>
+        private static object CreateDetailedCorrelatedEventReport(FlareEvent flare, CMEEvent cme)
+        {
+            var timeDifference = (cme.StartTime?.Subtract(flare.BeginTime ?? DateTime.MinValue).TotalMinutes) ?? 0;
+            var flareRiseDuration = (flare.PeakTime?.Subtract(flare.BeginTime ?? DateTime.MinValue).TotalMinutes) ?? 0;
+
+            return new
+            {
+                FlareID = flare.FlrID,
+                FlareClassType = flare.ClassType,
+                FlareBeginTime = flare.BeginTime,
+                FlarePeakTime = flare.PeakTime,
+                FlareRiseDuration = flareRiseDuration,
+                FlareLink = flare.Link,
+                CMEID = cme.ActivityID,
+                CMEStartTime = cme.StartTime,
+                TimeDifferenceMins = timeDifference,
+                CMEAnalyses = cme.CMEAnalyses.Select(analysis => new
+                {
+                    analysis.IsMostAccurate,
+                    analysis.Type,
+                    analysis.Speed,
+                    analysis.Latitude,
+                    analysis.Longitude,
+                    analysis.HalfAngle,
+                    analysis.Link
+                })
+            };
+        }
+
+        /// <summary>
+        ///  Find solar events that might have occurred nearly at the same time
+        /// </summary>
+        /// <param name="FLRdata"></param>
+        /// <param name="CMEdata"></param>
+        /// <returns> List of objects</returns>
+
+        public List<object> FindDetailedCorralatedEvents(List<object> FLRdata, List<object> CMEdata)
+        {
+
+            var flareEvents = FLRdata.Cast<FlareEvent>().Where(f => f.BeginTime.HasValue && f.PeakTime.HasValue).ToList();
+            var CMEEvents = CMEdata.Cast<CMEEvent>().Where(c => c.StartTime.HasValue).ToList();
+
+
+            TimeSpan timeVariation = TimeSpan.FromMinutes(5); // 5 minutes for a slightly wider window
+            TimeSpan maxFlareDuration = TimeSpan.FromHours(2); // Maximum flare duration
+
+            var correlatedEvents = flareEvents
+                .SelectMany(flare => CMEEvents
+                    .Where(cme => flare.BeginTime.HasValue && flare.PeakTime.HasValue && cme.StartTime.HasValue &&
+                                  (cme.StartTime.Value >= flare.BeginTime.Value - timeVariation &&
+                                   cme.StartTime.Value <= flare.PeakTime.Value + timeVariation) &&
+                                  (flare.PeakTime.Value - flare.BeginTime.Value) <= maxFlareDuration)
+                    .Select(cme => CreateDetailedCorrelatedEventReport(flare, cme)))
```

Comment:
For things like this, I'd recommend breaking stuff out into separate variables to make it a bit more readable and understandable at a glance. I'm guessing this just looking for events that happened within a time frame, but it would probably take me a few minutes to actually figure out exactly what this expression is doing.